src/lib/artefact.ml

```diff
@@ -1,90 +1,83 @@
 
+(* This file is free software, part of FTW. See file "LICENSE" for more information *)
+
+(* Artefact kinds *)
+(* ************************************************************************* *)
+
+type yan_variant =
+  | Single
+  | With_bonus
+  | List of { criterion : string list; }
+
+type kind =
+  | Bonus
+  | Ranking
+  | Yes_Alt_No of yan_variant
+
+(* Artefact values *)
+(* ************************************************************************* *)
+
+type bonus = int
```

Comment:
A few reasons:
1) storing integers is more compact (integers take between 1 and 8 bytes depending on the magnitude whereas a float will always take 8 byte since it's stored as a double)
2) bonus are only used in case of equality, and therefore any value that can be compared (to establish a ranking) can be used as bonus, a float between 0 and 1 is quite restrictive and only serves to further the notion that all rankings are established based upon a one-dimensional integer score, to which we would add the bonus, which is not really a good way to think about it I'd say
3) integers are much easier to input, store, manipulate and display accurately than floats (for instance, `0.1` and `0.2` cannot be stored exactly as floats, and thus we would need to rely on rounding when printing to ensure that we do not print weird things such as `0.100000001` in some situations)